ui/src/dashboards/components/variablesControlBar/VariablesControlBar.tsx

```diff
@@ -0,0 +1,129 @@
+// Libraries
+import React, {PureComponent} from 'react'
+import {connect} from 'react-redux'
+import _ from 'lodash'
+
+// Components
+import VariableDropdown from 'src/dashboards/components/variablesControlBar/VariableDropdown'
+import {EmptyState, ComponentSize} from 'src/clockface'
+
+// Styles
+import 'src/dashboards/components/variablesControlBar/VariablesControlBar.scss'
+
+// Actions
+import {selectValue} from 'src/variables/actions'
+
+// Types
+import {AppState} from 'src/types/v2'
+import {VariablesState} from 'src/variables/reducers'
+
+// Decorators
+import {ErrorHandling} from 'src/shared/decorators/errors'
+
+interface OwnProps {
+  dashboardID: string
+}
+
+interface StateProps {
+  variables: VariablesState
+}
+
+interface DispatchProps {
+  selectValue: typeof selectValue
+}
+
+type Props = StateProps & DispatchProps & OwnProps
+
+@ErrorHandling
+class VariablesControlBar extends PureComponent<Props> {
+  render() {
+    // return <div className="variables-control-bar" />
+    if (_.isEmpty(this.variablesForDashboard)) {
+      return (
+        <div className="variables-control-bar">
+          <EmptyState
+            size={ComponentSize.ExtraSmall}
+            className="variables-control-bar--empty"
+          >
+            <EmptyState.Text text="To see variable controls here, use a variable in a cell query" />
+          </EmptyState>
+        </div>
+      )
+    }
+
+    return (
+      <div className="variables-control-bar">
+        {Object.keys(this.variablesForDashboard).map(variableID => {
+          const initialSelected = _.get(
+            this.valuesForDashboard,
+            `${variableID}.selectedValue`
+          )
+          const name = _.get(
+            this.variablesForDashboard,
+            `${variableID}.variable.name`,
+            ''
+          )
+          const values = _.get(
+            this.valuesForDashboard,
+            `${variableID}.values`,
+            []
+          )
+
+          return (
+            <VariableDropdown
+              key={variableID}
+              variableID={variableID}
+              name={name}
+              values={values}
+              initialSelected={initialSelected}
+              onSelect={this.handleSelectValue}
+            />
+          )
+        })}
+      </div>
+    )
+  }
+
+  private get variablesForDashboard() {
+    const {variables} = this.props.variables
+
+    const variableIDs = Object.keys(this.valuesForDashboard)
+    let variablesForDash = {}
+
+    Object.keys(variables).forEach(variableID => {
```

Comment:
I think this stuff would belong better in a "selector" utility function. A selector is a function that reads Redux state and transforms/filters it into a nice format for a component to use. Selectors get used in the `mstp` for a component. That way the component stays focused only on presentation, and the selection logic can be reused across components.

[Here](https://github.com/influxdata/influxdb/blob/ccc8165b168d7db823091ed0d02f61a4a27a70ca/ui/src/timeMachine/selectors/index.ts#L13-L18) is an example of a simple selector and [where it gets used](https://github.com/influxdata/influxdb/blob/ccc8165b168d7db823091ed0d02f61a4a27a70ca/ui/src/timeMachine/components/Queries.tsx#L128-L130).

A lot of people also [memoize selectors](https://github.com/reduxjs/reselect) for performance reasons, but it's probably a detail not worth bothering with at the moment.